target/linux/ath79/dts/ar7100_mikrotik_routerboard-4xx.dtsi

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-1.0-or-later OR MIT
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7100.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &led_user;
+		led-failsafe = &led_user;
+		led-running = &led_user;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user: user {
+			label = "mikrotik:yellow:user";
+			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		led1 {
+			label = "mikrotik:green:led1";
+			gpios = <&cpld 0 GPIO_ACTIVE_LOW>;
+		};
+
+		led2 {
+			label = "mikrotik:green:led2";
+			gpios = <&cpld 2 GPIO_ACTIVE_LOW>;
+		};
+
+		led3 {
+			label = "mikrotik:green:led3";
+			gpios = <&cpld 1 GPIO_ACTIVE_LOW>;
+		};
+
+		led4 {
+			label = "mikrotik:green:led4";
+			gpios = <&cpld 3 GPIO_ACTIVE_LOW>;
+		};
+
+		led5 {
+			label = "mikrotik:green:led5";
+			gpios = <&cpld 8 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	nand: nand {
+		compatible = "mikrotik,rb4xx-nand";
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	ngpios = <31>;
+};
+
+&pcie0 {
+	// NOT TESTED (n.b. ath79_pci_irq struct has the slot/pin/IRQ mappings if needed)
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	compatible = "mikrotik,rb4xx-spi";
+
+	flash@0 {
+		compatible = "pm25lv512", "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
```

Comment:
Please use the new routerbootpart driver

Reply:
I've just tested, and pushed it

Reply:
```
root@OpenWrt:/# cat /proc/mtd 
dev:    size   erasesize  name
mtd0: 00010000 00001000 "partitions"
mtd1: 0000b000 00001000 "routerboot1"
mtd2: 00001000 00001000 "hard_config"
mtd3: 00001000 00001000 "bios"
mtd4: 00002000 00001000 "routerboot2"
mtd5: 00001000 00001000 "soft_config"
mtd6: 00040000 00020000 "booter"
mtd7: 003c0000 00020000 "kernel"
mtd8: 07c00000 00020000 "ubi"

```